src/core/net/ip6.cpp

```diff
@@ -374,6 +392,50 @@ otError Ip6::SendDatagram(Message &aMessage, MessageInfo &aMessageInfo, IpProto
     if (aMessageInfo.GetPeerAddr().IsMulticast() &&
         aMessageInfo.GetPeerAddr().GetScope() > Address::kRealmLocalScope)
     {
+        if (GetInstance().GetThreadNetif().GetMle().IsMulticastChildrenSubscribed(header.GetDestination()))
```

Comment:
I don't know if Thread specifies this, but my first thought would be to use other aproach. Here if let's say packet is destined to ff05::1 which is registered by one of our SED, we would add two IPv6 packets to the queue:

1) One without MPL header, destined to ff05::1
2) One with MPL header and IP-in-IP header, destined to FF03::FC

Personally, i would prefer to not modify this part of ip6.cpp, and allow to create IP-in-IP packet. However, i would modify mesh-forwarder part related to forwarding multicast message to SEDs. And instead, check IPv6 address included in the inner IPv6 header, if the outer IPv6 header destination address is set to FF03::FC.

SEDs must be subscribed to FF03::FC address anyway:

> To forward multicast packets that have scope larger than realm-local, all Thread devices receive and process MPL Data Messages sent to the realm-local ALL_MPL_FORWARDERS address.

I guess if this approach doesn't violate specification, it would remove a need for changes in the ip6.cpp file which simplifies the overall code.

Reply:
Thank you for creating the issue to the Thread Group for discussion.

Personally, i wouldn't drop MPL header and outer IPv6 header, as we don't do it in the Thread 1.1.1 certification, when we ping SEDs using Realm-Local All Thread Nodes multicast address. But i agree, we can wait for clarification, and i see a huge save in number of octets sent over the air,